### PR TITLE
FII-030 - Display the Inline files on the File inspector - UI implementation (LEAN-920)

### PR DESCRIPTION
**Pull request by asouqi**
In this PR: will get data of inline files from ModelMap and will update design [figma](https://www.figma.com/file/vspHhzKTbwSkZWQWOcmIw2/Lean-workflow---Final-UI-views?node-id=3807%3A1)

### CODE REVIEW

**Comment by atyponci** [comment]
Can one of the admins verify this patch?

File: src/components/FileManager/InlineFilesSection.tsx [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+    submissionId: string,
+    name: string,
+    file: File,
+    typeId: string
+  ) => Promise<boolean>
+  handleDownload: (url: string) => void
+  dispatch: Dispatch<Action>
+}> = ({ submissionId, handleReplace, handleDownload, modelMap, dispatch }) => {
+  const inlineFiles = useMemo(() => {
+    const files: {
```
Comment (mbartenev-atypon) [inline src/components/FileManager/InlineFilesSection.tsx:69]: can we move that function out into the lib to improve readability of that code, or maybe a custom hook?

**Comment by asouqi** [comment]
@mbartenev-atypon moved to lib

File: src/components/FileManager/InlineFilesSection.tsx [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+  handleReplace: (
+    submissionId: string,
+    name: string,
+    file: File,
+    typeId: string
+  ) => Promise<boolean>
+  handleDownload: (url: string) => void
+  dispatch: Dispatch<Action>
+}> = ({ submissionId, handleReplace, handleDownload, modelMap, dispatch }) => {
+  const inlineFiles = getInlineFiles(modelMap)
```
Comment (mbartenev-atypon) [inline src/components/FileManager/InlineFilesSection.tsx:52]: cool. Can we still use useMemo here?

**Review by mbartenev-atypon — APPROVED** [review]

**Comment by mbartenev-atypon** [comment]
Approving with small notes.

File: yarn.lock [hunk 3]
```diff
@@ -1581,6 +1620,15 @@
     deepmerge "^3.2.0"
     js-beautify "^1.9.0"
 
+"@manuscripts/manuscripts-json-schema@^1.49.5":
```
[the remark this answers is not in the document]
Comment (asouqi) [reply yarn.lock]: I move it to `dependencies`, check now yearn.lock

File: src/components/FileManager/InlineFilesSection.tsx [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+  console.log(inlineFiles, modelMap)
```
[the remark this answers is not in the document]
Comment (asouqi) [reply src/components/FileManager/InlineFilesSection.tsx]: oh, I forgot to remove it

File: src/components/FileManager/InlineFilesSection.tsx [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+  externalFile,
+  submissionId,
+  handleReplace,
+  handleDownload,
+  dispatch,
+}) => {
+  const { isOpen, toggleOpen, wrapperRef } = useDropdown()
+
+  if (!externalFile || !externalFile.filename) {
+    return <></>
```
[the remark this answers is not in the document]
Comment (asouqi) [reply src/components/FileManager/InlineFilesSection.tsx]: update

File: src/components/FileManager/InlineFilesSection.tsx (hunk 2 above)
Comment (asouqi) [reply src/components/FileManager/InlineFilesSection.tsx]: updated to use Memo

**Pull request merged**